src/ProxyClient/HttpDispatcher.php

```diff
@@ -187,52 +192,65 @@ protected function getServers()
      */
     private function fanOut(RequestInterface $request)
     {
-        $requests = [];
-
-        $uri = $request->getUri();
-
-        // If a base URI is configured, try to make partial invalidation
-        // requests complete.
-        if ($this->baseUri) {
-            if ($uri->getHost()) {
-                // Absolute URI: does it already have a scheme?
-                if (!$uri->getScheme() && '' !== $this->baseUri->getScheme()) {
-                    $uri = $uri->withScheme($this->baseUri->getScheme());
-                }
-            } else {
-                // Relative URI
-                if ('' !== $this->baseUri->getHost()) {
-                    $uri = $uri->withHost($this->baseUri->getHost());
-                }
-
-                if ($this->baseUri->getPort()) {
-                    $uri = $uri->withPort($this->baseUri->getPort());
+        /** @var RequestInterface[] $requests */
+        $requests = [$request];
```

Comment:
we do not use that in the case of having baseUris. i would prefer to structure the code like this:
```php
if (0 !== \count) {
    $requests = $this->requestToBaseUris($request);
} else {
    $requests = [$request];
}
```

Reply:
Done.